packages/remix-dev/vite/node/adapter.ts

```diff
@@ -1,69 +1,23 @@
-// @ts-nocheck
-// adapted from https://github.com/solidjs/solid-start/blob/ccff60ce75e066f6613daf0272dbb43a196235a4/packages/start/node/fetch.js
-import { once } from "events";
-import { type IncomingMessage, type ServerResponse } from "http";
-import multipart from "parse-multipart-data";
+import type {
+  IncomingHttpHeaders,
+  IncomingMessage,
+  ServerResponse,
+} from "node:http";
+import { once } from "node:events";
+import { Readable } from "node:stream";
 import { splitCookiesString } from "set-cookie-parser";
-import { Readable } from "stream";
-import { File, FormData, Headers, Request as BaseNodeRequest } from "undici";
-import { type ServerBuild, installGlobals } from "@remix-run/node";
+import {
+  type ServerBuild,
+  installGlobals,
+  createReadableStreamFromReadable,
+} from "@remix-run/node";
 import { createRequestHandler as createBaseRequestHandler } from "@remix-run/server-runtime";
 
-installGlobals();
-
-function nodeToWeb(nodeStream) {
-  let destroyed = false;
-  let listeners = {};
-
-  function start(controller) {
-    listeners["data"] = onData;
-    listeners["end"] = onData;
-    listeners["end"] = onDestroy;
-    listeners["close"] = onDestroy;
-    listeners["error"] = onDestroy;
-    for (let name in listeners) nodeStream.on(name, listeners[name]);
-
-    nodeStream.pause();
-
-    function onData(chunk) {
-      if (destroyed) return;
-      controller.enqueue(chunk);
-      nodeStream.pause();
-    }
-
-    function onDestroy(err) {
-      if (destroyed) return;
-      destroyed = true;
-
-      for (let name in listeners)
-        nodeStream.removeListener(name, listeners[name]);
+import invariant from "../../invariant";
 
-      if (err) controller.error(err);
-      else controller.close();
-    }
-  }
-
-  function pull() {
-    if (destroyed) return;
-    nodeStream.resume();
-  }
-
-  function cancel() {
-    destroyed = true;
-
-    for (let name in listeners)
-      nodeStream.removeListener(name, listeners[name]);
-
-    nodeStream.push(null);
-    nodeStream.pause();
-    if (nodeStream.destroy) nodeStream.destroy();
-    else if (nodeStream.close) nodeStream.close();
-  }
-
-  return new ReadableStream({ start: start, pull: pull, cancel: cancel });
-}
+installGlobals();
```

Comment:
Thanks for proceeding with this PR!

I was wondering about what's the future plan of this forced `installGlobals` during `vite dev`.
Node 18 ships `fetch`, `Request`, etc... globals, so I thought forcing polyfills here might not be appropriate and better to be opt-in choice by individual users.

Looking at what happens in other adapters https://github.com/remix-run/remix/pull/7009 (and also `unstable-vite-express` currently has explicit `installGlobals` in `server.mjs`), the general direction seems to remove automatic/forced `installGlobals`.

Currently this polyfills kicks in whenever remix vite plugin is loaded, so I think it's not even possible for custom server (during dev) to opt-out from `installGlobals` currently.

Here is a quick repro:

https://stackblitz.com/edit/remix-run-remix-ws1uig?file=vite.config.ts

```tsx
//
// vite.config.ts
//

let fetch1 = fetch;

import { unstable_vitePlugin as remix } from '@remix-run/dev';
import { defineConfig } from 'vite';
import tsconfigPaths from 'vite-tsconfig-paths';

let fetch2 = fetch;

export default defineConfig({
  clearScreen: false,
  plugins: [remix(), tsconfigPaths()],
});

let fetch3 = fetch;

// the log happens twice in parent and child compilers
//   parent: true false
//   child:  true true
console.log(fetch1 === fetch2, fetch1 === fetch3);
```

Reply:
Yep, I was wanting to follow this up separately. Thanks for looking into this!